### PR TITLE
chore: sync agent skills and add CLAUDE.md patterns

### PR DESCRIPTION
**Pull request by WhatIfWeDigDeeper**
## Summary
- Syncs \`learn\`, \`peer-review\`, and \`pr-comments\` agent skills from upstream (\`whatifwedigdeeper/agent-skills\`) with updated hashes in \`skills-lock.json\`
- Adds new \`CLAUDE.md\` pattern: fix vulnerabilities across all packages in one pass (using \`rg\` with proper glob exclusion to cover nested packages like \`lambda-api/cdk/\`)
- Extends lambda-api pagination, \`void asyncFn()\`, and Zustand stale-state notes with additional edge cases discovered during recent sessions
- Clarifies \`npm --prefix\` flag ordering gotcha for subagent use

## Test Plan
- [ ] No code changes — verify CI passes (lint, build, tests unaffected)
- [ ] Confirm skills-lock.json hashes match upstream skill content

---
🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: CLAUDE.md [hunk 1]
```diff
@@ -89,6 +89,8 @@ When **updating**: use the `update-deps` skill (npm only), then run the full val
 
 **`audit:ci:all` stops at first failure** — To find "Consider not allowlisting advisories" messages across all packages, run audit-ci per-package individually (`for dir in ...; do (cd "$dir" && npx -y audit-ci --config .auditconfig.json); done`) rather than `npm run audit:ci:all`, which halts at the first failing package and hides stale-allowlist warnings in later packages.
 
+**Fix vulnerabilities across all packages in one pass** — When a CI audit failure names a specific package (e.g. `drizzle-orm`), grep for it across all `package.json` files before committing the fix: `grep -r '"drizzle-orm"' */package.json`. Multiple packages often share the same vulnerable dependency; fixing only the one CI reported causes a second CI failure on the next package in the chain. After applying all fixes, run `npm run audit:ci:all` (or the per-package loop) to confirm every package passes before committing.
```
[the remark this answers is not in the document]
Comment (WhatIfWeDigDeeper) [reply CLAUDE.md]: Good catch —  with  only matches one directory deep, missing nested packages like `lambda-api/cdk/package.json`. Switched to `rg --glob '!node_modules/**' --glob '**/package.json'` in 93798e5, which also aligns with the existing CLAUDE.md guidance to prefer `rg` over `grep`.

---
🤖 Generated with [Claude Code](https://claude.com/claude-code)

File: CLAUDE.md [hunk 2]
```diff
@@ -208,7 +210,8 @@ Prefer individual CRUD operations (`addStage`, `updateStage`, `removeStage`) ove
 - **`tsx` IPC in sandbox**: `npx tsx` requires a Unix socket for hot-reload IPC which is blocked in sandbox; use `dangerouslyDisableSandbox: true` or `node --import tsx/esm` as alternative
 - **`docs/types/lambda-api/api.mermaid` is hand-maintained**: `ts-to-mermaid` cannot resolve `zod` (runtime import, not a type-level dependency) — the `docs:types:lambda-api` script was removed. Update the mermaid file manually when `api.ts` types change.
 - **Mermaid erDiagram syntax gotchas**: `PK`/`FK`/`UK` are reserved attribute key constraint tokens — use `PartitionKey`/`SortKey` for DynamoDB keys (lowercase `pk`/`sk` also fail; multi-letter names like `GSI1PK` are fine). Avoid `|` inside quoted annotations (write `0or1`). `nullable().optional()` → `type|null?`; `.optional()` only → `type?`. classDiagram enum values with spaces need quotes (`"given offer"`); hyphenated values work unquoted.
```
[the remark this answers is not in the document]
Comment (WhatIfWeDigDeeper) [reply CLAUDE.md:212]: The schema docs (`docs/schema/lambda-api/README.md`) prescribe the correct approach (use `PartitionKey`/`SortKey`), while `CLAUDE.md` serves as a gotcha list — the two documents aren't inconsistent, they serve different purposes. The lowercase `pk`/`sk` failure was empirically observed during the original implementation session: Mermaid's erDiagram parser treats these tokens case-insensitively, so `pk`/`sk` as attribute names also get parsed as key constraint markers. Keeping both the uppercase and lowercase warnings helps future developers avoid the same pitfall.

---
🤖 Generated with [Claude Code](https://claude.com/claude-code)

File: .agents/skills/pr-comments/references/report-templates.md [hunk 3]
```diff
@@ -14,6 +14,9 @@ Use these templates for Step 14, omitting lines that don't apply.
 {poll status — omit if not applicable}
 
 [List of each action taken]
+
+{closing line}
+<PR URL>
 ```
```
[the remark this answers is not in the document]
Comment (WhatIfWeDigDeeper) [reply .agents/skills/pr-comments/references/report-templates.md:20]: Thanks for the catch — you're right that `{closing line}` + `<PR URL>` being separate in the template while the MANDATORY definition shows them as a 2-line block creates ambiguity. This file lives in `.agents/skills/` which is maintained upstream at `WhatIfWeDigDeeper/agent-skills`; per this repo's Agent Skills Policy, reviewer suggestions on those files are addressed upstream rather than applied directly here. I'll carry this clarification forward in the next upstream sync.

---
🤖 Generated with [Claude Code](https://claude.com/claude-code)

File: .agents/skills/pr-comments/references/report-templates.md [hunk 4]
```diff
@@ -60,6 +63,9 @@ Updated PR body: reflects 3 commits (was 1).
 Exited: no new threads after iteration 3.
 
 M out-of-scope declined comments — file follow-up issues? [all/select/none]
+
+{closing line}
+<PR URL>
 ```
```
[the remark this answers is not in the document]
Comment (WhatIfWeDigDeeper) [reply .agents/skills/pr-comments/references/report-templates.md:69]: Same situation as the standard template thread — this is a valid observation about potential duplication between `{closing line}` and the explicit `<PR URL>` line. This file is maintained upstream at `WhatIfWeDigDeeper/agent-skills`; per the Agent Skills Policy it won't be edited directly here. Will carry this forward in the next upstream sync.

---
🤖 Generated with [Claude Code](https://claude.com/claude-code)
